Assets/Mirror/Runtime/Authentication/Authenticator.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.ComponentModel;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Mirror
+{
+    // UnityEvent definitions
+    [Serializable] public class UnityEventConnection : UnityEvent<NetworkConnection> { }
+
+    public abstract class Authenticator : MonoBehaviour
+    {
+        /// <summary>
+        /// The current Authenticator used by Mirror. 
+        /// </summary>
+        public static Authenticator activeAuthenticator;
+
+        /// <summary>
+        /// Determines if client has been authenticated
+        /// </summary>
+        /// <returns>True if a client has been authenticated successfully</returns>
+        public abstract bool IsAuthenticated();
```

Comment:
I think this flag needs to move to NetworkConnection.

If you are a server and you have 10 connection but only 5 are authenticated,  then what would this flag have?